src/dbSta/src/dbSta.tcl

```diff
@@ -112,5 +112,29 @@ proc sta_warn { id msg } {
   utl::warn STA $id $msg
 }
 
+define_cmd_args "replace_design" {instance module}
+
+proc replace_design { instance module } {
+  set design [get_design_error $module]
+  if { $design != "NULL" } {
+    set inst [find_hier_inst_cmd $instance]
+    replace_design_cmd $inst $design
+    return 1
+  }
+  return 0
+}
+
+proc get_design_error { arg } {
```

Comment:
Strange name as it has nothing to do with design.  Perhaps get_module?

Reply:
"design" was chosen to denote some hierarchical nature of the module.   UIs that I had in mind were

- get_designs
- remove_designs (not implemented yet)
- replace_design

replace_module seems a little too similar with replace_cell and doesn't convey a hierarchical module swapping.

Reply:
It is confusing against repair_design/report_design_area/triton_part_design/current_design/link_design/... which operate on the whole design not a submodule module. 

replace_hierarchical_module ?

Reply:
UIs may need a few iterations to converge.   Can we merge this and work on improving UIs later?